src/index.tsx

```diff
@@ -209,10 +211,20 @@ export default class ReactToPrint extends React.Component<IReactToPrintProps> {
     public render() {
         const {
             trigger,
+            onBeforeGetContent
         } = this.props;
 
         return React.cloneElement(trigger(), {
-            onClick: this.handlePrint,
+            onClick: () => {
+                const onBeforeGetContentPromise = onBeforeGetContent();
```

Comment:
Also, you need to check that `onBeforeGetContent` is passed before trying to call it. This should be similar to https://github.com/gregnb/react-to-print/blob/67df094343115d35f002ef66a752abb3a6b9d261/src/index.tsx#L55